mino/minogrpc/mod.go

```diff
@@ -144,15 +145,23 @@ func WithRandom(r io.Reader) Option {
 }
 
 // NewMinogrpc creates and starts a new instance. it will try to listen for the
-// address and returns an error if it fails.
-func NewMinogrpc(addr net.Addr, router router.Router, opts ...Option) (*Minogrpc, error) {
-	socket, err := net.Listen(addr.Network(), addr.String())
+// address and returns an error if it fails. "listen" is the local address,
+// while "public" is the public node address. If public is empty it uses the
+// local address.
+func NewMinogrpc(listen net.Addr, public string, router router.Router, opts ...Option) (*Minogrpc, error) {
+	socket, err := net.Listen(listen.Network(), listen.String())
 	if err != nil {
 		return nil, xerrors.Errorf("failed to bind: %v", err)
 	}
 
+	if public == "" {
+		public = socket.Addr().String()
+	}
+
+	dela.Logger.Info().Msgf("public address is: %s", public)
+
 	tmpl := minoTemplate{
-		myAddr: session.NewAddress(socket.Addr().String()),
+		myAddr: session.NewAddress(public),
```

Comment:
Don't you want to make sure that `public` is a valid address here? It's user-input, so I'd verify it before putting it in a structure.